### PR TITLE
Hdf5 For Windows V2

### PR DESCRIPTION
**Pull request by jpopelar**
Remake of PR #29826 due to branch conflicts and duplicated commits

### CODE REVIEW

File: var/spack/repos/builtin/packages/sqlite/package.py [hunk 1]
```diff
@@ -48,6 +48,8 @@ class Sqlite(AutotoolsPackage):
     variant('dynamic_extensions', default=True, description='Support loadable extensions')
     variant('rtree', default=True, description='Build with Rtree module')
 
+    # This isn't ideal but since we don't have a "not" style syntax we have to condition
+    # the depends_on based on the actual platform vs. the spec's platform
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/sqlite/package.py]: This looks vestigial (or some accompanying changes did not make it over)

File: var/spack/repos/builtin/packages/perl/package.py [hunk 2]
```diff
@@ -329,8 +328,8 @@ def _setup_dependent_env(self, env, dependent_spec, deptypes):
         if perl_lib_dirs:
             perl_lib_path = ':'.join(perl_lib_dirs)
             env.prepend_path('PERL5LIB', perl_lib_path)
-        if is_windows:
-            env.append_path('PATH', self.prefix.bin)
+            if is_windows:
+                env.append_path('PATH', self.prefix.bin)
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/perl/package.py]: Was something going wrong when you didn't make this change?

I ask because I think `prefix.bin` will always be added to PATH (this is performed for build/run dependencies in build/run contexts, respectively), so I'm surprised these lines existed in the first place.

File: var/spack/repos/builtin/packages/netlib-lapack/package.py [hunk 3]
```diff
@@ -82,6 +85,8 @@ class NetlibLapack(CMakePackage):
     depends_on('blas', when='+external-blas')
     depends_on('netlib-xblas+fortran+plain_blas', when='+xblas')
     depends_on('python@2.7:', type='test')
+    depends_on('ninja@1.10.0:', when='platform=windows')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/netlib-lapack/package.py]: Note `CMakePackage` has `depends_on(ninja)`, so unless `netlib-lapack` specifically needs a newer `ninja` version than other CMake packages, it shouldn't be necessary to add this.

File: lib/spack/spack/build_systems/windows_variants.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.directives import variant
+from spack.package import Package
+
+
+class WindowsPackage(Package):
```
Comment (chuckatkins) [inline lib/spack/spack/build_systems/windows_variants.py]: Is this class actually handling *windows* options or *msvc* options? I could be misunderstanding this but there seems to be an assumption that windows = msvc which I don't believe is an assumptions spack should be making.

File: lib/spack/llnl/util/symlink.py [hunk 5]
```diff
@@ -54,7 +53,16 @@ def _win32_junction(path, link):
...
+    command = "mklink"
+    default_args = [link, path]
+    if os.path.isdir(path):
+        # try using a junction
+        default_args.insert(0, '/J')
+    else:
+        # try using a hard link
+        default_args.insert(0, '/H')
+
+    Executable(command)(*default_args)
```
Comment (scheibelp) [inline lib/spack/llnl/util/symlink.py]: This undoes: https://github.com/spack/spack/commit/88b1bf751d76e010584b251f482cdfb0ba5351a1

Is there a reason for that?

File: var/spack/repos/builtin/packages/bzip2/package.py [hunk 6]
```diff
@@ -125,7 +148,9 @@ def install(self, spec, prefix):
                 for libname in (lib, lib1, lib2):
                     symlink(lib3, libname)
 
-        with working_dir(prefix.bin):
-            force_remove('bunzip2', 'bzcat')
-            symlink('bzip2', 'bunzip2')
-            symlink('bzip2', 'bzcat')
+        # These files won't be in a Windows installation
+        if sys.platform != 'win32':
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/bzip2/package.py]: This should also use `platform=windows` like above.

File: var/spack/repos/builtin/packages/hdf5/package.py [hunk 7]
```diff
@@ -379,10 +382,15 @@ def cmake_args(self):
         ]
 
         api = spec.variants['api'].value
+        if sys.platform == 'win32' and self.spec.satisfies('+staticmt'):
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/hdf5/package.py]: I'm wondering if `hdf5~shared` and `hdf5+staticmt` are redundant: can you make a static library for `hdf5` on Windows that doesn't use this?

File: var/spack/repos/builtin/packages/zlib/package.py [hunk 8]
```diff
@@ -3,16 +3,16 @@
...
 import os
 
+from spack.build_systems.windows_variants import WindowsPackage
 from spack.package import *
 
 
-class Zlib(Package):
+# Although zlib comes with a configure script, it does not use Autotools
+# The AutotoolsPackage causes zlib to fail to build with PGI
+class Zlib(Package, WindowsPackage):
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/zlib/package.py]: The only thing `WindowsPackage` provides is `staticmt`. Since this package doesn't use that variant here, does it in fact need to inherit `WindowsPackage`?

**Review by scheibelp — CHANGES_REQUESTED** [review]
Let me know if any of these requests seem to miss the point.

File: var/spack/repos/builtin/packages/hdf5/package.py [hunk 9]
```diff
@@ -86,15 +88,16 @@ class Hdf5(CMakePackage):
             values=('default', 'v114', 'v112', 'v110', 'v18', 'v16'),
             multi=False)
 
-    depends_on('cmake@3.12:', type='build')
-
-    depends_on('mpi', when='+mpi')
+    if sys.platform != 'win32':
```
Comment (byrnHDF) [inline var/spack/repos/builtin/packages/hdf5/package.py]: There is a MSMPI available for windows. The spack package is next in review.

File: var/spack/repos/builtin/packages/hdf5/package.py (hunk 7 above)
Comment (byrnHDF) [reply var/spack/repos/builtin/packages/hdf5/package.py]: In this day and age, Windows "mt" should stop being used. Windows 10+ has improved runtime libraries.

File: var/spack/repos/builtin/packages/perl/package.py (hunk 2 above)
Comment (jpopelar) [reply var/spack/repos/builtin/packages/perl/package.py]: I'll look into it. spack install perl is hanging for me on Windows so there's something more here

File: lib/spack/llnl/util/symlink.py (hunk 5 above)
Comment (jpopelar) [reply lib/spack/llnl/util/symlink.py]: Not that I can see, maybe got missed by the rebase. Should be fixed now

File: var/spack/repos/builtin/packages/hdf5/package.py (hunk 7 above)
Comment (jpopelar) [reply var/spack/repos/builtin/packages/hdf5/package.py]: IIRC, the "regular" build links nonshared with /MD flags, so there is a reasonable variant to be had here

File: var/spack/repos/builtin/packages/hdf5/package.py (hunk 9 above)
Comment (johnwparent) [reply var/spack/repos/builtin/packages/hdf5/package.py]: Draft MSMPI Spack package can be found [here](https://github.com/spack/spack/pull/30228). Updates are upcoming pending support for the WinWDK/SDK packages in Spack (also in progress).

**Comment by scheibelp** [comment]
@jpopelar can you push an empty commit here (see also: https://github.com/spack/spack/pull/31589#issuecomment-1192209684).

File: var/spack/repos/builtin/packages/sqlite/package.py [hunk 10]
```diff
@@ -48,8 +48,9 @@ class Sqlite(AutotoolsPackage):
     variant('dynamic_extensions', default=True, description='Support loadable extensions')
     variant('rtree', default=True, description='Build with Rtree module')
 
-    depends_on('readline')
-    depends_on('zlib')
+    if not self.spec.satisfies('platform=windows'):
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/sqlite/package.py]: This is causing https://github.com/spack/spack/runs/7472966301?check_suite_focus=true

Comment (scheibelp) [reply var/spack/repos/builtin/packages/sqlite/package.py]: (there is no `self` here)

File: var/spack/repos/builtin/packages/msmpi/package.py [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    provides('mpi')
+
+    conflicts('platform=linux')
+    conflicts('platform=darwin')
+    conflicts('platform=cray')
+
+    def url_for_version(self, version):
+        return 'https://download.microsoft.com/download/A/E/0/AE002626-9D9D-448D-8197-1EA510E297CE/msmpisetup.exe'
+
+    def install(self):
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/msmpi/package.py]: This will probably cause problems because Spack will always call `Package.install()` with arguments. The typical signature is `def install(self, spec, prefix):`

**Comment by jpopelar** [comment]
Just pushed up a commit to trim out windows/msvc_variants and addressed your feedback @scheibelp. I'll let the tests go again and LMK if you notice anything else out of the ordinary!

**Comment by scheibelp** [comment]
https://github.com/spack/spack/issues/31265 looks related to the style failure, if applying the fix described there works, then could you mention that in that issue?

(one thing that's strange: that issue describes how this would come up for non-Ubuntu systems, but I checked the runner for this style check and it appears to be Ubuntu, so I'm not sure how this would come up there - it is the same error message though)

**Comment by jpopelar** [comment]
77796a6 does not look like it did it; just applied @alalazo's suggestion to bootstrap.py by adding a `.local` into the `bin_dir` path. Are there any other issues that discuss this topic?

**Comment by jpopelar** [comment]
@scheibelp Pinging you as per your requests. I'll look into the conflicts if you want to examine anything else

File: lib/spack/llnl/util/symlink.py [hunk 12]
```diff
@@ -54,8 +51,6 @@ def _win32_junction(path, link):
         path = os.path.join(parent, path)
         path = os.path.abspath(path)
 
-    CreateHardLink(link, path)
```
Comment (scheibelp) [inline lib/spack/llnl/util/symlink.py]: Removing this line means this function no longer does anything. Is something going wrong when it does this?

File: var/spack/repos/builtin/packages/hdf5/package.py [hunk 13]
```diff
@@ -86,15 +87,16 @@ class Hdf5(CMakePackage):
...
+        depends_on('mpi', when='+mpi')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/hdf5/package.py]: Since this PR includes an MPI implementation for Windows, can we allow depends_on('mpi', when=+mpi) for Windows as well? If not (i.e. if this PR will be merged without that) then this should add conflicts('+mpi', when='platform=windows'), and this if can also be removed in that case).

File: var/spack/repos/builtin/packages/hdf5/package.py [hunk 14]
```diff
@@ -86,15 +87,16 @@ class Hdf5(CMakePackage):
...
     depends_on('java', type=('build', 'run'), when='+java')
-    depends_on('szip', when='+szip')
+    # numactl does not currently build on darwin
+    if sys.platform != 'darwin' and sys.platform != 'win32':
+        depends_on('numactl', when='+mpi+fortran')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/hdf5/package.py]: Have you noticed that `hdf5` on Linux requires `numactl`? I just want to make sure since this PR is generally oriented at building `hdf5` on Windows, but if you noticed an additional constraint for `hdf5` on Linux then that could make sense to add here to (although it might be ideal to split that off into another PR).

File: var/spack/repos/builtin/packages/hdf5/package.py (hunk 7 above)
Comment (scheibelp) [reply var/spack/repos/builtin/packages/hdf5/package.py]: I don't think there's a `+staticmt` variant anymore, so I think this `if` needs to be refined.

**Review by scheibelp — COMMENTED** [review]
I have a few questions/requests. I'll do a more-thorough review once this is updated with the latest develop.

**Comment by scheibelp** [comment]
Close/reopen to restart tests

File: var/spack/repos/builtin/packages/hdf5/package.py (hunk 13 above)
Comment (jpopelar) [reply var/spack/repos/builtin/packages/hdf5/package.py]: I think to be on the safe side I'll make this a conflicts statement and then see how David Alexander's PR goes along. Will be in next commit

**Comment by jpopelar** [comment]
@scheibelp PTAL when convenient. Windows unit tests are passing but hitting failures with clingo and Mac

File: lib/spack/spack/test/llnl/util/file_list.py [hunk 15]
```diff
@@ -52,6 +65,15 @@ def header_list():
     return h
 
 
+plat_static_ext = "lib" if is_windows else "a"
+
+
+plat_shared_ext = "dll" if is_windows else ".so"
```
Comment (johnwparent) [inline lib/spack/spack/test/llnl/util/file_list.py]: @jpopelar This line is causing (at least some of) your current CI failures. The extra `.` character in the `".so"` designation for `plat_shared_ext` is causing the string comp for the library names to fail.

Comment (jpopelar) [reply lib/spack/spack/test/llnl/util/file_list.py]: Just pushed a fix, thanks!

Comment (johnwparent) [reply lib/spack/spack/test/llnl/util/file_list.py]: This fix is already present in the RPath PR, #31930 btw. A lot of the changes from that PR seem to be present in this one, so you may want to check out that PR for future issues.

File: var/spack/repos/builtin/packages/netlib-lapack/package.py [hunk 16]
```diff
@@ -105,6 +105,7 @@ class NetlibLapack(CMakePackage):
     depends_on("blas", when="+external-blas")
     depends_on("netlib-xblas+fortran+plain_blas", when="+xblas")
     depends_on("python@2.7:", type="test")
+    depends_on("cmake@3.15:")
```
Comment (johnwparent) [inline var/spack/repos/builtin/packages/netlib-lapack/package.py]: Is this true for all versions of lapack?

**Comment by jpopelar** [comment]
@spackbot run pipeline

**Comment by scheibelp** [comment]
@spackbot run pipeline

File: pyproject.toml [hunk 17]
```diff
@@ -1,6 +1,6 @@
 [tool.black]
 line-length = 99
-target-version = ['py27', 'py35', 'py36', 'py37', 'py38', 'py39', 'py310']
+target-version = ['py27', 'py35', 'py36', 'py37', 'py38']
```
Comment (scheibelp) [inline pyproject.toml]: This should still test Python 3.9/3.10

File: var/spack/repos/builtin/packages/zlib/package.py [hunk 18]
```diff
@@ -3,15 +3,14 @@
...
-# Although zlib comes with a configure script, it does not use Autotools
-# The AutotoolsPackage causes zlib to fail to build with PGI
 import glob
 import os
 
 from spack.package import *
 
 
+# Although zlib comes with a configure script, it does not use Autotools
+# The AutotoolsPackage causes zlib to fail to build with PGI
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/zlib/package.py]: IMO this should be reverted (it doesn't look like there were any changes other than moving a comment)

File: var/spack/repos/builtin/packages/sqlite/package.py [hunk 19]
```diff
@@ -202,6 +204,25 @@ def configure_args(self):
...
 
+    def configure(self, spec, prefix):
+        if not self.spec.satisfies("platform=windows"):
+            super(Sqlite, self).configure(spec, prefix)
+
+    def build(self, spec, prefix):
+        if self.spec.satisfies("platform=windows"):
+            nmake = Executable("nmake.exe")
+            print(self.configure_flag_args)
+            nmake('CC = "%s"' % os.environ.get("SPACK_CC"), "Makefile.msc")
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/sqlite/package.py]: Did you find this was necessary? We should already be setting CC in the environment: it generally points to the Spack compiler wrapper e.g. `<spack-prefix>/lib/spack/env/cc` and `SPACK_CC` points to the actual compiler path  - in that case we wouldn't get any of the extra `-L` flags inserted by Spack.

File: var/spack/repos/builtin/packages/perl/package.py [hunk 20]
```diff
@@ -368,8 +368,8 @@ def _setup_dependent_env(self, env, dependent_spec, deptypes):
         if perl_lib_dirs:
             perl_lib_path = ":".join(perl_lib_dirs)
             env.prepend_path("PERL5LIB", perl_lib_path)
-        if is_windows:
-            env.append_path("PATH", self.prefix.bin)
+            if is_windows:
+                env.append_path("PATH", self.prefix.bin)
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/perl/package.py]: I see this is moving this, but I'm surprised this is needed: we should already be adding `<prefix>/bin/` to the `PATH` automatically. Does the installation on windows work without this?

**Review by scheibelp — CHANGES_REQUESTED** [review]
This looks great although I have a few suggestions for cleanup. Also, I don't see any changes to hdf5 itself: does `spack install hdf5+mpi` work on Windows without any further configuration? I would have expected we'd have to add a Windows-specific configuration which encouraged choosing msmpi on Windows (I assume other MPI implementations would fail at the moment).

File: var/spack/repos/builtin/packages/msmpi/package.py [hunk 21]
```diff
@@ -0,0 +1,34 @@
...
+    Microsoft Support Team
+    """
+
+    homepage = "https://www.microsoft.com/en-us/download/default.aspx"
+    maintainers = ["jpopelar"]
+
+    version(
+        "10.0",
+        sha256="7dae13797627726f67fab9c1d251aec2df9ecd25939984645ec05748bdffd396",
+        extension="exe",
```
Comment (johnwparent) [inline var/spack/repos/builtin/packages/msmpi/package.py:22]: Does this work? AFAIK this should fail because we'll still try to extract the extension and fail (because it's a binary). Should this be set to `expand=False`?

File: var/spack/repos/builtin/packages/sqlite/package.py [hunk 22]
```diff
@@ -202,6 +204,25 @@ def configure_args(self):
 
         return args
 
+    def configure(self, spec, prefix):
+        if not self.spec.satisfies("platform=windows"):
+            super(Sqlite, self).configure(spec, prefix)
+
+    def build(self, spec, prefix):
+        if self.spec.satisfies("platform=windows"):
+            nmake = Executable("nmake.exe")
```
Comment (johnwparent) [inline var/spack/repos/builtin/packages/sqlite/package.py]: This is no longer necessary, `nmake` should be set by the build environment setup.

File: pyproject.toml (hunk 17 above)
Comment (jpopelar) [reply pyproject.toml]: Sorry, that was me experimenting with my black installation. Reverted now

File: var/spack/repos/builtin/packages/hdf5/package.py [hunk 23]
```diff
@@ -196,13 +196,16 @@ class Hdf5(CMakePackage):
...
     depends_on("java", type=("build", "run"), when="+java")
     depends_on("szip", when="+szip")
     depends_on("zlib@1.1.2:")
 
     # The compiler wrappers (h5cc, h5fc, etc.) run 'pkg-config'.
-    depends_on("pkgconfig", type="run")
+    # Further, check for standard mpi in this same Windows-excluding loop
+    for plat in ['cray', 'darwin', 'linux']:
+        depends_on("pkgconfig", when="platform=%s" % plat, type="run")
+        depends_on("mpi", when="+mpi, platform=%s" % plat)
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/hdf5/package.py]: You can actually keep `depends_on(mpi, when=+mpi)` and do not need to make `depends_on(mpi)` platform-specific: the idea is that every platform needs to build the MPI virtual with `+mpi`, but Windows requires the additional constraint that you specifically need `msmpi`. 

(this loop is still required though for the pkgconfig dependency)

File: var/spack/repos/builtin/packages/msmpi/package.py [hunk 24]
```diff
@@ -0,0 +1,42 @@
...
+        return "https://download.microsoft.com/download/A/E/0/AE002626-9D9D-448D-8197-1EA510E297CE/msmpisetup.exe"
+
+    def determine_version(self, exe):
+        output = Executable("mpiexec.exe")
+        ver_str = re.search("\[Version ([0-9.]+)\]", output)
+        return Version(ver_str.group(0)) if ver_str else None
+
+    def install(self, spec, prefix):
+        installer = Executable("msmpisetup.exe")
+        installer("-unattend", "-installroot", prefix)
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/msmpi/package.py]: At one point I think you mentioned `-installroot` wasn't working as expected (i.e. the installer was ignoring it and installing to some common location) - is that still true?

Furthermore, with `-unattend`, does this require user interaction when the install fails (e.g. when `msmpi` is already installed)?

File: var/spack/repos/builtin/packages/sqlite/package.py [hunk 25]
```diff
@@ -4,6 +4,7 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 import os
 import re
+import sys
 from tempfile import NamedTemporaryFile
 
 import spack.platforms
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/sqlite/package.py]: It just occurred to me: `sqlite` is an `AutotoolsPackage`, and I thought that class defined a conflict with `platform=windows`. Is `sqlite` needed as a dependency of `hdf5`? If not, then we could move it to another PR. If you want to keep it, I would want to make sure we can install/concretize it. We could leverage multi-build-system support to make it easier to install this on Windows in that case.

File: lib/spack/spack/detection/path.py [hunk 26]
```diff
@@ -234,7 +234,6 @@ def by_executable(packages_to_check, path_hints=None):
                 exe_pattern_to_pkgs[exe].append(pkg)
         # Add Windows specific, package related paths to the search paths
         path_hints.extend(compute_windows_program_path_for_package(pkg))
-
```
Comment (scheibelp) [inline lib/spack/spack/detection/path.py]: This looks vestigial (no modifiers to this file other than whitespace)

**Review by scheibelp — CHANGES_REQUESTED** [review]
I few additional requests. Let me know if you disagree with any

File: var/spack/repos/builtin/packages/msmpi/package.py (hunk 24 above)
Comment (jpopelar) [reply var/spack/repos/builtin/packages/msmpi/package.py]: For the first point, I mentioned that the installation was empty even with the option. I eventually got around this by declaring msmpi as a non-buildable external in my packages.yaml

Pursuant to the second point, there is no user input required during installation failure, as I've tested

File: var/spack/repos/builtin/packages/sqlite/package.py (hunk 25 above)
Comment (jpopelar) [reply var/spack/repos/builtin/packages/sqlite/package.py]: I'll be phasing the work on sqlite out in my next commit; stay tuned

File: lib/spack/spack/detection/path.py (hunk 26 above)
Comment (jpopelar) [reply lib/spack/spack/detection/path.py]: Probably leftover damage from some debug output; fixed now

File: var/spack/repos/builtin/packages/sqlite/package.py (hunk 25 above)
Comment (johnwparent) [reply var/spack/repos/builtin/packages/sqlite/package.py]: Can you repurpose those commits into a separate PR? Sqlite support is something I'm very interested in (and assumed this PR would cover) and if you don't have the cycles to do the extra work, I can/will happily take over/add it to my mega package PR.

File: var/spack/repos/builtin/packages/msmpi/package.py (hunk 24 above)
Comment (johnwparent) [reply var/spack/repos/builtin/packages/msmpi/package.py]: I can confirm behavior on my local setup as well.

**Comment by jpopelar** [comment]
While I wait on the GitLab CI, can anyone give me details on bootstrap/macos-gnupg-sources? I'm not familiar with what goes into this set of the tests

File: var/spack/repos/builtin/packages/msmpi/package.py (hunk 24 above)
Comment (scheibelp) [reply var/spack/repos/builtin/packages/msmpi/package.py]: > I eventually got around this by declaring msmpi as a non-buildable external in my packages.yaml

I'm not clear how that addresses the issue of being unable to specify the Spack prefix. If I understand: you have confirmed that this does in fact install `msmpi` when it has not been installed yet - did I get that right? If so, the docstring for the package should mention that `msmpi` installs to a predetermined location.

**Comment by scheibelp** [comment]
> While I wait on the GitLab CI, can anyone give me details on bootstrap/macos-gnupg-sources? I'm not familiar with what goes into this set of the tests

The error `[2022-11-15-17:44:47.766694] Error: Failed to install gnupg due to TimeoutError: The read operation timed out` does not look related to your changes, so I'm inclined to re-run it. To do that I'll need to close and reopen the PR.

**Comment by scheibelp** [comment]
@spackbot run pipeline

**Comment by jpopelar** [comment]
@spacbot run pipeline

**Comment by jpopelar** [comment]
@spackbot run pipeline

**Review by scheibelp — APPROVED** [review]

**Comment by scheibelp** [comment]
Thanks @jpopelar!

**Comment by jpopelar** [comment]
You're welcome @scheibelp, have a good Thanksgiving

File: var/spack/repos/builtin/packages/hdf5/package.py [hunk 27]
```diff
@@ -196,13 +196,16 @@ class Hdf5(CMakePackage):
...
     depends_on("mpi", when="+mpi")
     depends_on("java", type=("build", "run"), when="+java")
     depends_on("szip", when="+szip")
     depends_on("zlib@1.1.2:")
 
     # The compiler wrappers (h5cc, h5fc, etc.) run 'pkg-config'.
-    depends_on("pkgconfig", type="run")
+    # Skip this on Windows since pkgconfig is autotools
+    for plat in ["cray", "darwin", "linux"]:
+        depends_on("pkgconfig", when="platform=%s" % plat, type="run")
```
Comment (skosukhin) [inline var/spack/repos/builtin/packages/hdf5/package.py:208]: I'm late with my comments but nonetheless.
1. I disagree with the comment: `pkgconfig` is not `autotools`.
2. The compiler wrappers will not work without `pkg-config` in the `PATH`.

I understand that the compiler wrappers of `hdf5` a rarely used but if `pkg-config` works on Windows, the code above should be reverted. Otherwise, the wrappers should be removed after the installation to avoid confusion.
